snapdtool/cmdutil.go

```diff
@@ -112,9 +112,33 @@ func CommandFromSystemSnap(name string, cmdArgs ...string) (*exec.Cmd, error) {
 	cmdPath := filepath.Join(root, name)
 
 	if from == "snapd" {
-		// We do not need to specify the linker or library
-		// path from files coming from the snapd snap.
-		return exec.Command(cmdPath, cmdArgs...), nil
+		// check if the elf interpreter invoked by the binary will work
+		// without any tweaks, which is true if the snaps are mounted at
+		// /snap, otherwise we need to set up a command to invoke it
+		// directly
+		// TODO perhaps check the symlink right away
+		slashSnapPrefix := filepath.Join(dirs.GlobalRootDir, "snap") + "/"
+		if strings.HasPrefix(root, slashSnapPrefix) {
```

Comment:
Like I have commented on the origin PR. I would like this check to be changed to also verify that the /snap -> /var/lib/snapd/snap symlink  exists. In that case, even if the path does not start with /snap, we should *not* call through the dynamic linker.

Reply:
https://github.com/canonical/snapd/pull/14375/commits/e0039edf491c5e7646b43b47a6c345564519c01b